### PR TITLE
Fix f90_correct tests

### PR DESCRIPTION
**Pull request by bryanpkc**
`test/f90_correct` contains a number of compile-time tests which use FileCheck to verify the correctness of the LLVM IR produced by Flang. They were not being run during `make check-all`. This patch moves them into a sub-directory, teaches llvm-lit to run them, and fix a few issues in the tests themselves.

### CODE REVIEW

**Comment by kiranchandramohan** [comment]
Thanks @bryanpkc for this PR. This PR addresses issue https://github.com/flang-compiler/flang/issues/899.

I would like to suggest a restructuring of the test directory.
1) PR https://github.com/flang-compiler/flang/pull/910 which we are currently considering to merge brings in a directory called debug_info in the top level of the flang test directory. Can you move the debug_module_import.f90 there?
2) Can all the tests which check LLVM IR (daz.f90, f-i8.f90, f-i8.f90, f-noi8.f90, f-nor8.f90, f-r8.f90, save.f90, shift64.f90) be moved to an llvm_ir_correct directory in the top level?
3) Can the other tests be also put in a suitable top-level directory?

**Comment by bryanpkc** [comment]
@kiranchandramohan Thanks for your comments and pointing me to #899 and #900, which I didn't have a chance to read. I will rebase my PR on that one, and make the suggested changes.

**Comment by bryanpkc** [comment]
Refactored as suggested and rebased on master.

**Comment by bryanpkc** [comment]
Pinging reviewers.

File: tools/flang1/flang1exe/semfin.c [hunk 1]
```diff
@@ -539,7 +539,7 @@ semfin(void)
 
   /* Process data from SAVE statements */
 
-  if (sem.savloc)
+  if (sem.savloc || sem.savall)
```
Comment (kiranchandramohan) [inline tools/flang1/flang1exe/semfin.c:542]: For @gklimowicz's reference. This bug was introduced in commit https://github.com/flang-compiler/flang/commit/d2d64f083b004dc215c052dd22c3108008ee46da. @bryanpkc's change corrects the code here.

Also, note that the same issue reported in your forum.
https://forums.developer.nvidia.com/t/msave-broken-in-ce19-10/136441

**Comment by kiranchandramohan** [comment]
Once @shivaramaarao approves @bryanpkc can merge.

**Review by shivaramaarao — COMMENTED** [review]
I am seeing following two failures after pulling in these changes

Failing Tests (2):
    Flang :: debug_info/debug_module_import.f90
    Flang :: llvm_ir_correct/daz.f90

can these be fixed?

**Comment by gklimowicz** [comment]
@shivaramaarao what's the build/test environment where you're running into these test failures? I'm assuming x86_64 and LLVM 9? I didn't see them on Power, but maybe I'm not trying hard enough. :-)

**Comment by bryanpkc** [comment]
@shivaramaarao These tests run successfully with LLVM 10. They might not work with LLVM 9 or older. I can try to relax the FileCheck patterns to see if that makes them work with LLVM 9.

**Comment by bryanpkc** [comment]
I have verified that these tests pass with the release_90 branches of flang-compiler/flang-driver and flang-compiler/llvm, on x86_64. They also pass with the release_100 branch of flang-compiler/classic-flang-llvm-project. @shivaramaarao Could you provide more information on the failures you are seeing? Which branch are you building?

**Comment by shivaramaarao** [comment]
> I have verified that these tests pass with the release_90 branches of flang-compiler/flang-driver and flang-compiler/llvm, on x86_64. They also pass with the release_100 branch of flang-compiler/classic-flang-llvm-project. @shivaramaarao Could you provide more information on the failures you are seeing? Which branch are you building?

my tests were on release_90 branches of lang-compiler/flang-driver and flang-compiler/llvm, on x86_64. I will cross check these failures again.

**Comment by shivaramaarao** [comment]
> > I have verified that these tests pass with the release_90 branches of flang-compiler/flang-driver and flang-compiler/llvm, on x86_64. They also pass with the release_100 branch of flang-compiler/classic-flang-llvm-project. @shivaramaarao Could you provide more information on the failures you are seeing? Which branch are you building?
> 
> my tests were on release_90 branches of lang-compiler/flang-driver and flang-compiler/llvm, on x86_64. I will cross check these failures again.

I rechecked and the errors were because the patch was not applied correctly for above two files. all the tests pass in my environment as well.

**Review by shivaramaarao — APPROVED** [review]

**Comment by bryanpkc** [comment]
Thanks @shivaramaarao .

**Pull request merged**